### DISCUSSION TITLE
Loading button image from file path string

### DISCUSSION THREAD

**Discussion by itslyn** [AvaloniaUI/Avalonia#15574] [Q&A] [answered]
I'm trying to set the Image source for a button from a string that contains a file path located somewhere on my computer.

Here is my AXAML:
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/65033d19-163b-4c06-9051-e44fae2590fe)

Here is my view model code:
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/5a606ced-535e-4778-a5c6-afb755d994ac)

The IconPath string is "file:///D:/Pictures/216.png"

**Comment by maxkatz6** [comment, marked as answer]
Make IconImage a Bitmap type:
```
public Bitmap IconImage { get; set; }
```

then you can create a bitmap like:
```
IconImage = new Bitmap("D:/Pictures/216.png"); // (adjust the code to use IconPath)
```

**Reply by maxkatz6** [reply]
Alternatively, you can keep it a `string`, but implement a `IValueConverter` that would convert that.

**Reply by itslyn** [reply]
Thank you! Do you also know how I can get the button's image to have rounded corners? My button has CornerRadius="5" but the image inside still has sharp edges

**Reply by stevemonaco** [reply]
You should clip it using a border. eg.
```xml
<Button CornerRadius="80">
  <Border CornerRadius="80" ClipToBounds="True">
    <Image Source="{Binding IconImage}" />
  </Border>
</Button>
```

**Reply by itslyn** [reply]
Thank you! I had the border with the corner radius but didn't put the ClipToBounds
